src/app/shared/components/addscenario/addscenario.component.ts

```diff
@@ -168,9 +170,30 @@ export class AddScenarioModal implements OnInit, OnDestroy {
         });
     }
 
+    public equationCheck(check){
+        if (check == true) {
```

Comment:
Not a mandatory change, but since `check` is a boolean, you can just say `if (check) {` here